shared/src/main/java/alcatraz/shared/Lobby.java

```diff
@@ -6,42 +6,51 @@
 package alcatraz.shared;
 
 import java.io.Serializable;
-import java.util.Map;
+import java.rmi.RemoteException;
+import java.util.HashMap;
 
 public class Lobby implements Serializable {
-    private long id;
-    private Map<String, Player> players;
-    private String ownerId;
-    // TODO: add secret
+    public final long id;
+    private HashMap<String, Player> players;
+    public final String ownerId;
+    private final String secret;
+    private static final long MAXSIZE = 4;
 
-    public Lobby(long id, Map<String, Player> players, String ownerId) {
+    public Lobby(long id, String ownerId, String secretToken) {
         this.id = id;
-        this.players = players;
         this.ownerId = ownerId;
+        this.secret = secretToken;
     }
 
     public long getId() {
         return this.id;
     }
 
-    public Map<String, Player> getPlayers() {
+    public HashMap<String, Player> getPlayers() {
         return this.players;
     }
 
     public String getOwner() {
         return this.ownerId;
     }
 
-    public boolean addPlayer(Player player) {
-        if (this.players.size() < 4) {
+    public void addPlayer(Player player) throws RemoteException {
```

Comment:
The error could be more specific, the RemoteException is thrown when something is wrong with RMI connection, so it can be even misleading.
I think in this case we should create a new error e.g. LobbyIsFull or something like that.

Reply:
see LobbyFullException